app/lib/hesa/client.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Hesa
+  class Client
+    AUTH_URL = "https://identity.hesa.ac.uk/Account/RemoteLogOn?ReturnDomain=https://datacollection.hesa.ac.uk&ReturnUrl=%2f"
+
+    class LoginError < StandardError; end
+
+    class UnauthorizedError < StandardError; end
+
+    def login
+      page = agent.get(AUTH_URL)
+      form = page.form_with(id: "loginForm")
+      form.EmailAddress = Settings.hesa.username
+      form.Password = Settings.hesa.password
+      page = form.submit
+
+      unless logged_in?(page)
+        raise(LoginError, "Unable to login to HESA with form")
+      end
+
+      page
+    end
+
+    def logged_in?(page)
+      !!(page.css("a").text.match("Log off") && page.css("a").text.match(Settings.hesa.username))
```

Comment:
I don't think we should check the page content for this. I don't think we can check for an 'UnauthorizedError' on 37 either as we're not getting proper HTTP codes back. I'd get rid of this check (which means you can get rid of the relevant bits of the spec and fixture)